### PR TITLE
Add ws-security support and REST basic auth handler to the micro integrator profile

### PR DESCRIPTION
**Pull request by Arshardh**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/callback/AbstractPasswordCallback.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+import org.wso2.carbon.user.core.claim.ClaimManager;
+import org.wso2.carbon.user.core.profile.ProfileConfigurationManager;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+
```
Comment (milindaperera) [inline components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/callback/AbstractPasswordCallback.java]: Let's add class comment

File: components/org.wso2.carbon.micro.integrator.security/pom.xml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                        </Import-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
+                    </instructions>
+                </configuration>
+            </plugin>
+
+        </plugins>
+    </build>
+
+</project>
```
Comment (ThishaniLucas) [inline components/org.wso2.carbon.micro.integrator.security/pom.xml]: Shall we fix this EOF?

File: components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/callback/AbstractPasswordCallback.java (hunk 1 above)
Comment (Arshardh) [reply components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/callback/AbstractPasswordCallback.java]: Done

File: components/org.wso2.carbon.micro.integrator.security/pom.xml (hunk 2 above)
Comment (Arshardh) [reply components/org.wso2.carbon.micro.integrator.security/pom.xml]: Done

File: components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/callback/RealmConfigXMLProcessor.java [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.security.callback;
+
+import org.apache.axiom.om.*;
```
[the remark this answers is not in the document]
Comment (Arshardh) [reply components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/callback/RealmConfigXMLProcessor.java]: Done

**Pull request merged**